tests/integration/README.md

```diff
@@ -0,0 +1,172 @@
+
+## Available Fixtures
+
+See `conftest.py` for fixture details:
+- `integration_env`: Full integration environment (backend + services)
+- `dashboard_backend`: Just the backend without services
+- `monitor_services`, `detector_services`, `reduction_services`: Specific service combinations
+
+## Test Helpers
+
+See `helpers.py` for synchronous waiting utilities (e.g., `wait_for_data`, `wait_for_condition`). Avoid using `time.sleep()`—use helpers instead.
+
+## Writing Integration Tests
+
+### Basic Pattern
+
+```python
+@pytest.mark.integration
+@pytest.mark.services('monitor')
+def test_my_workflow(integration_env):
+    backend = integration_env.backend
+
+    # Create workflow with unique identifier for filtering
+    workflow_id = WorkflowId(
+        instrument='dummy',
+        namespace='monitor_data',
+        name='monitor_histogram',
+        version=1,
+    )
+
+    # Start workflow
+    backend.workflow_controller.start_workflow(workflow_id, ['monitor1'], config)
+
+    # Wait for data using helper, filtering by workflow_id
+    def check_for_job_data():
+        backend.update()
+        for job_number, source_data in backend.job_service.job_data.items():
+            # Only check jobs belonging to OUR workflow
+            if backend.job_service.job_info.get(job_number) == workflow_id:
+                if 'monitor1' in source_data:
+                    return True
+        return False
+
+    wait_for_condition(check_for_job_data, timeout=10.0)
+
+    # Filter workflow-specific jobs for assertions
+    workflow_jobs = [
+        job_num for job_num, wf_id in backend.job_service.job_info.items()
+        if wf_id == workflow_id
+    ]
+    assert len(workflow_jobs) > 0, f"Expected at least one job for {workflow_id}"
+
+    # Clean up
+    backend.workflow_controller.stop_workflow(workflow_id)
+```
+
+### Writing Tests Robust to Fixture Scope Changes
+
+**IMPORTANT**: Currently, integration test fixtures create new backend processes for every test (function scope). In the future, fixtures may be changed to session or module scope to improve performance by sharing processes across multiple tests.
+
+Write tests that work regardless of whether fixtures are isolated or shared:
+
+#### DO: Filter by Test-Specific Identifiers
+
+Always filter global state by your test's workflow ID or source name before making assertions:
+
+```python
+@pytest.mark.integration
+@pytest.mark.services('monitor')
+def test_my_workflow(integration_env: IntegrationEnv):
+    backend = integration_env.backend
+    workflow_id = WorkflowId(
+        instrument='dummy',
+        namespace='monitor_data',
+        name='monitor_histogram',
+        version=1,
+    )
+
+    backend.workflow_controller.start_workflow(workflow_id, ['monitor1'], config)
+
+    # ✅ GOOD: Filter by workflow_id before checking
```

Comment:
If this is a good pattern, it should be provided by the framework. We shouldn't duplicate this complex code across tests.